File: comparison_Python_R_piechart.py
#!/usr/bin/python
#####################################################################
# Author:		Alberto Albz Marocchino
# Date:			19-02-2017
# Purpose:  comparison between python and R :: pie plot
# Source:   python
#####################################################################

#--- import packages ---#
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt

#--- generate sample---#
data = {'fruit':['apple','peach','melon'], 'percentage':[25,25,50]}
df   = pd.DataFrame(data)

#--- plot ---#
%matplotlib inline
ax1 = plt.subplot(111, aspect='equal')
df.plot(kind='pie', y='percentage', ax=ax1, autopct='%1.1f%%', startangle=90, shadow=False, labels=df['fruit'], legend = False, fontsize=14)

File: comparison_Python_R_piechart.r
#!/usr/bin/python
#####################################################################
# Author:		Alberto Albz Marocchino
# Date:			19-02-2017
# Purpose:     comparison between python and R :: cake plot
# Source:       python
#####################################################################

#--- generate sample---#
df <- data.frame(
  group = c("apple", "peach", "melon"),
  value = c(25, 25, 50)
  )

library(ggplot2)

pie <- ggplot(df, aes(x="", y=value, fill=group)) + 
       geom_bar(width = 1, stat = "identity") + theme_void() +
       coord_polar("y",start=0) +
       geom_text(aes(x=1, y = c(12.5,87.5,50), label=value))

pie


